Library/Homebrew/diagnostic.rb

```diff
@@ -832,6 +833,23 @@ def check_homebrew_prefix
         EOS
       end
 
+      def check_deleted_formula
+        kegs = Keg.all
+        deleted_formulae = []
+        kegs.each do |keg|
+          keg_name = keg.name
+          deleted_formulae << keg_name if Formulary.tap_paths(keg_name).size != 1
```

Comment:
```suggestion
          deleted_formulae << keg_name if Formulary.tap_paths(keg_name).blank?
```

as otherwise if it's in more than one tap it'll show as deleted